orcoursetrion/actions/github.py

```diff
@@ -45,6 +45,53 @@ def create_export_repo(course, term, description=None):
     return repo
 
 
+def rerun_studio(course, term, new_term, description=None):
+    """Run any actions needed to re-run a Studio course.
+
+    This will remove the hooks from the specified term, and then create a new
+    export repo for the ``new_term``.  It finds the repo based on
+    :py:const:`~orcoursetrion.config.ORC_STUDIO_ORG` as the organization,
+    :py:const:`~orcoursetrion.config.ORC_COURSE_PREFIX` as the prefix,
+    replacing all the dots in ``course`` and appending "-``term``".
+
+    Args:
+        course (str): Course name to be used to name repo (i.e. 6.004r)
+        term (str): Term the course is last run (i.e. 2015_Spring)
+        term (str): Term the course is expected to run again (i.e. 2018_Spring)
+        description (str): Optional description for repo to show up on github
+    Raises:
+        requests.RequestException
+        orcoursetrion.lib.GitHubUnknownError
+    Returns:
+        dict: Github dictionary of the newly created repo
+                (https://developer.github.com/v3/repos/#create)
+
+    """
+    github = GitHub(config.ORC_GH_API_URL, config.ORC_GH_OAUTH2_TOKEN)
+
+    # Find and clean up the old
+    repo_name = '{prefix}-{course}-{term}'.format(
+        prefix=config.ORC_COURSE_PREFIX,
+        course=course.replace('.', ''),
+        term=term
+    )
+    github.delete_web_hooks(config.ORC_STUDIO_ORG, repo_name)
```

Comment:
It gets really murky with the different stacks, and I haven't fully fleshed that out in my head.  Currently this whole design is based around just our RP stack, and this rerun thing would probably be bad to run for courses with multiple deploy points.

Thinking about it more, it may not even make sense to delete hooks at all, since the "rerun" action will probably occur during the semester and they may be working on both courses at the same time....I would say maybe we should just delete the staging hooks too, but I have concerns about that since at some point we won't want production getting old courses added due to a commit on an old repo (something that has occurred in the past).

I think for now it is probably best to leave this as is(to keep rerun simple) and treat reruns like a move operation, but in reality they are probably more of a subtle copy operation and we'll have to address that. To do that, we would need to split the `rerun` action into a `rerun with production still active` and a second `stop running the current/old version` action...